docs/reference/speaker.md

```diff
@@ -111,8 +111,11 @@ Finally, the boolean `loop` argument defines if the sound will be played only on
 
 It is possible to change the volume, pitch, balance, and loop parameters of a sound currently playing by calling again the `wb_speaker_play_sound` function with the same speakers and `sound` arguments.
 
-> **Note**: The path to the sound file should be defined either absolutely or relatively.
-If defined relatively, it will be searched first relatively to the robot controller folder.
+> **Note**: The path to the sound file should be defined either absolutely or relatively.<br>
+If the file is found by the controller using the provided path, it is loaded by the controller
+and the contents is streamed to Webots.<br>
+If the file is not found by the controller, only the given path is passed to Webots.
+If this path is relative, it will be searched first relatively to the robot controller folder.
 If not found there and if the robot is a PROTO, it will be searched relatively to the PROTO folder of the robot.
```

Comment:
We usually do not use any `<br>` tags in the documentation.
This paragraph is a bit unclear to me. I would suggest this if I understood well:
```suggestion
> **Note**: The path to the sound file should be defined either absolutely or relatively.
If this path is relative, it will be searched first relatively to the robot controller folder.
If the file is found by the controller using the provided path, it is loaded by the controller
and the contents is streamed to Webots.
If the file is not found by the controller, only the given path is passed to Webots and if the robot is a PROTO, it will be searched relatively to the PROTO folder of the robot.
```

Reply:
To me that's still a little unclear. I'd suggest this. (Let me know if I'm misunderstanding the original intent of the section.)
``` suggestion
> **Note**: The path to the sound file can be defined either absolutely or relatively.
If the path is relative, it will first be searched by the the robot controller, relative to its working directory.
If the file is found by the controller using the provided path, it is loaded by the controller and the contents are streamed to Webots.
If the file is not found by the controller, the given path is passed to Webots.
Webots will first search for the path relative to the robot controller folder.
If the file is not found relative to the controller folder and the robot is a PROTO, Webots will be search for the sound file relative to the folder of the PROTO file.
```

Reply:
Ok, not using br's anymore. Your suggestion above is not quite right.  It does not matter if the sound  file path is absolute or relative. If the file could be successfully opened by a controller (via fopen()), it is loaded by the controller and streamed.  This also supports usecases where the path is absolute, but the controller is on a different machine. The term "robot controller folder" is not really clear to me. If a relative path is given, it is used relative to the controller's **working** directory. 

I updated the markdown. Please check if you are satisfied with the new wording.

Reply:
Updated speaker.md again with suggestion from CoolSpy3 which came in in parallel to addressing omichel's feedback. 
Only difference is that the path is also used by controller if absolute:

"If the file path is accessible by the robot controller either absolutely or relative to its working directory, it is loaded by the controller and its contents are streamed to Webots."